### PR TITLE
Feature: filter results

### PR DESCRIPTION
**Pull request by EnQs**

### CODE REVIEW

File: src/main/java/enqs/worklog/filters/WorkdayFilterParameters.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package enqs.worklog.filters;
+
+@SuppressWarnings("unused")
+public class WorkdayFilterParameters {
+
+    String type;
+    String determinant;
+    String value;
```
Comment (Tyvrel) [inline src/main/java/enqs/worklog/filters/WorkdayFilterParameters.java]: private!!!!

File: src/main/java/enqs/worklog/filters/WorkdayFilterFactory.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    static public WorkdayFilter getWorkdayFilter(WorkdayFilterParameters parameters) {
+        return getWorkdayFilter(parameters.getType(), parameters.getDeterminant(), parameters.getValue());
+    }
+
+    static public WorkdayFilter getWorkdayFilter(String type, String determinant, String value) {
+        return filterType.valueOf(type).build(determinant, value);
+    }
+
+    @SuppressWarnings("unused")
+    private enum filterType {
```
Comment (Tyvrel) [inline src/main/java/enqs/worklog/filters/WorkdayFilterFactory.java]: `FilterType`

File: src/main/java/enqs/worklog/filters/WorkdayFilterFactory.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import java.time.Month;
+import java.util.function.BiFunction;
+
+public class WorkdayFilterFactory {
+
+    static public WorkdayFilter getWorkdayFilter(WorkdayFilterParameters parameters) {
+        return getWorkdayFilter(parameters.getType(), parameters.getDeterminant(), parameters.getValue());
+    }
+
+    static public WorkdayFilter getWorkdayFilter(String type, String determinant, String value) {
```
Comment (Tyvrel) [inline src/main/java/enqs/worklog/filters/WorkdayFilterFactory.java]: is this method used publicly?

File: src/main/java/enqs/worklog/filters/WorkdayFilterParameters.java (hunk 1 above)
Comment (EnQs) [reply src/main/java/enqs/worklog/filters/WorkdayFilterParameters.java]: ARGH... I have a problem... :P

File: src/main/java/enqs/worklog/WorkLogService.java [hunk 4]
```diff
@@ -14,8 +21,15 @@ public WorkLogService() {
         workLogRepository = new TextFileWorkLogRepository();
     }
 
-    public List<Workday> getWorkdays() {
-        return workLogRepository.find().getWorkdays();
+    //TODO: validation of parameters
+    public List<Workday> getWorkdays(List<WorkdayFilterParameters> parameters) {
+        ArrayList<WorkdayFilter> workdayFilters = parameters.stream()
+                .map(WorkdayFilterFactory::getWorkdayFilter)
+                .collect(Collectors.toCollection(ArrayList::new));
```
Comment (Tyvrel) [inline src/main/java/enqs/worklog/WorkLogService.java]: Collectors.toList()

**Pull request merged**